lib/routes/seagullApi.js

```diff
@@ -281,54 +281,75 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
 
           delete mergedUserPermissions[targetUserId];
           mergedUserPermissions = _.pickBy(mergedUserPermissions, function(p) { return userMatchesQueryOnPermissions(p, query); });
-          async.mapLimit(_.keys(mergedUserPermissions), 5, function(trustorUserId, callback) {
-            userApiClient.getUserInfo(trustorUserId, function(error, user) {
-              if (error) {
-                log.error(error, 'Error getting user for user id', trustorUserId);
-                return callback(error);
-              } else if (!user) {
-                log.error('No user returned for user id', trustorUserId);
-                return callback({statusCode: 500});
+          var userProfiles = [];
+
+          const mapLimit = util.promisify(async.mapLimit);
+          try {
+            const userIds = _.keys(mergedUserPermissions);
+            // Break requests for users into chunks of 200, so that the query parameter doesn't get too long
+            const results = await mapLimit(_.chunk(userIds, 200), 5, async usersChunk => {
+              const getUsers = util.promisify(userApiClient.getUsersWithIds);
+              try {
+                const users = await getUsers(usersChunk);
+                if (!users) {
+                  throw new Error(`No users returned for user id ${targetUserId}`);
+                } else {
+                  return users;
+                }
+              } catch (error) {
+                throw new Error(`Error getting users: ${error}`);
               }
+            });
+            userProfiles = _.flatten(results);
 
-              if (!userMatchesQueryOnUser(user, query)) {
-                return callback();
-              }
+            if (!_.isEqual(_.map(userProfiles, 'userid').sort(), _.sortBy(userIds))) {
```

Comment:
Just curious: assuming no duplicates, wouldn't a simple length check suffice here?

Reply:
You're quite right.